tests/test_plugin_loader_edge_cases.py

```
@@ -265,26 +266,38 @@ def test_load_plugins_from_directory_sys_path_manipulation_error(self):
+        Verify that get_custom_plugin_dirs returns local directories even when user dir creation fails.
         """
-        with patch("mmrelay.config.get_base_dir", return_value="/restricted"):
-            with patch("os.path.exists", return_value=True):
+        with patch(
+            "mmrelay.plugin_loader._get_plugin_root_dirs",
+            return_value=["/restricted/plugins"],
+        ):
+
+            def side_effect(path, **_kwargs):
+                if path.startswith("/restricted"):
+                    raise PermissionError("Permission denied")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)